api/turing/webhook/webhook.go

```diff
@@ -0,0 +1,74 @@
+package webhook
+
+import (
+	"context"
+
+	"github.com/caraml-dev/mlp/api/pkg/webhooks"
+)
+
+var (
+	OnRouterCreated    = webhooks.EventType("on-router-created")
+	OnRouterUpdated    = webhooks.EventType("on-router-updated")
+	OnRouterDeleted    = webhooks.EventType("on-router-deleted")
+	OnRouterDeployed   = webhooks.EventType("on-router-deployed")
+	OnRouterUndeployed = webhooks.EventType("on-router-undeployed")
+
+	OnRouterVersionCreated  = webhooks.EventType("on-router-version-created")
+	OnRouterVersionDeleted  = webhooks.EventType("on-router-version-deleted")
+	OnRouterVersionDeployed = webhooks.EventType("on-router-version-deployed")
+
+	OnEnsemblerCreated = webhooks.EventType("on-ensembler-created")
+	OnEnsemblerUpdated = webhooks.EventType("on-ensembler-updated")
+	OnEnsemblerDeleted = webhooks.EventType("on-ensembler-deleted")
+)
+
+var events = []webhooks.EventType{
+	OnRouterCreated,
+	OnRouterUpdated,
+	OnRouterDeleted,
+	OnRouterDeployed,
+	OnRouterVersionCreated,
+	OnRouterVersionDeleted,
+	OnRouterVersionDeployed,
+	OnRouterUndeployed,
+	OnEnsemblerCreated,
+	OnEnsemblerUpdated,
+	OnEnsemblerDeleted,
+}
+
+type webhook struct {
+	webhookManager webhooks.WebhookManager
+}
+
+type Client interface {
+	TriggerWebhooks(ctx context.Context, eventType webhooks.EventType, body interface{}) error
+}
+
+func NewWebhook(cfg *webhooks.Config) (Client, error) {
+	webhookManager, err := webhooks.InitializeWebhooks(cfg, events)
+	if err != nil {
+		return nil, err
+	}
+
+	return webhook{
+		webhookManager: webhookManager,
+	}, nil
+}
+
+func (w webhook) TriggerWebhooks(ctx context.Context, eventType webhooks.EventType, body interface{}) error {
+	if !w.isEventConfigured(eventType) {
+		return nil
+	}
+
+	return w.webhookManager.InvokeWebhooks(
+		ctx,
+		eventType,
+		body,
```

Comment:
Can we also add the event name in the body? I imagine for some webhooks it might be easier to have one endpoint for several events, then differentiate what to do based on `eventType`. Like with our sync use case, the webhook might simply put the request in a queue (so no need to have several endpoint), and there would be another worker that will process the queue later on (but it needs to know the `eventType`

Reply:
Yeah sure that's possible. Should we be making this change for Merlin too?

Reply:
Merlin already include the `eventType` in the payload body. It will be nice if it can be added by MLP side, but isn't it going to change the structure of the payload, like how do we insert new field to an `interface{}` variable? 🤔 Or we can modify Turing without changing MLP, it will be okay too for me

Reply:
> Merlin already include the eventType in the payload body.

Ah yes you're right I just saw that 😮  

> but isn't it going to change the structure of the payload, like how do we insert new field to an interface{} variable? 🤔

Hmm, what do you mean? I was thinking of just adding something like this (just as in Merlin) in the MLP package (assuming that Merlin hadn't already added the event type to the payload `p`):

```go
type WebhookRequest struct {
	EventType EventType     `json:"event_type"`
	Data      interface{} `json:"data"`
}

...

func (w *SimpleWebhookManager) InvokeWebhooks(
	ctx context.Context,
	event EventType,
	p interface{},
	onSuccess func([]byte) error,
	onError func(error) error,
) error {
	...
	originalPayload, err := json.Marshal(p)

	newRequestBody := &WebhookRequest{
		EventType: event,
		Data: p,
	}
	originalPayload, err := json.Marshal(newRequestBody)
	...
}

```

But yeah, since Merlin already includes the event type in the payload body, I'll just add it here to Turing without changing MLP then :D